client-linuxapp/src/main.rs

```diff
@@ -104,57 +156,154 @@ async fn get_client_list(rv_entry: &RendezvousInterpretedDirective) -> Result<Ve
     Ok(service_client_list)
 }
 
-async fn perform_to1(
+/// TO1: Sends HelloRV, Receives HelloRVAck
+async fn perform_hellorv(
     devcred: &dyn DeviceCredential,
     client: &mut ServiceClient,
-) -> Result<COSESign> {
-    log::trace!(
-        "Starting TO1 with credential {:?} and client {:?}",
-        devcred,
-        client
-    );
-
+) -> Result<COSESign, ClientError> {
     let sig_type = DeviceSigType::StSECP384R1;
 
-    // Send: HelloRV, Receive: HelloRVAck
     let hello_rv = messages::v11::to1::HelloRV::new(
         devcred.device_guid().clone(),
         SigInfo::new(sig_type, vec![]),
     );
     let hello_rv_ack: RequestResult<messages::v11::to1::HelloRVAck> =
         client.send_request(hello_rv, None).await;
-    let hello_rv_ack = hello_rv_ack.context("Error sending HelloRV")?;
+
+    let hello_rv_ack = match hello_rv_ack {
+        Ok(hello_rv_ack) => hello_rv_ack,
+        Err(e) => {
```

Comment:
Instead of this pattern match all the time, you could do:
```
let hello_rv_ack = hello_rv_ack
    .context("Error sending HelloRV")
    .map_err(|e| ClientError::Request(ErrorResult::new(
      ErrorCode::InternalServerError,
      String::from("Error sending HelloRV"),
      MessageType::TO1HelloRV,
      e,
))?;
```

This basically transforms (with `.context()`) from a `Result<_, fdo_http_wrapper::client::Error>` to `Result<_, anyhow::Error>`, and then (with `.map_err()`) to `Result<_, ClientError>`, and then with the `?` operator, it uses the normal "return if it errored" mechanism.